### PR TITLE
Replaced message parameters in GameSession with MessageContext

### LINKED ISSUE #433: Remove dependency on Discord Message object throughout code

### PR DESCRIPTION
**Pull request by taahamahdi**
Closes #433.

### CODE REVIEW

File: src/models/game_session.ts [hunk 1]
```diff
@@ -94,10 +98,13 @@ export default class GameSession {
      * Ends an active GameRound
      * @param guessed - Whether the round ended via a correct guess, or other (timeout, error, etc)
      */
-    async endRound(guessed: boolean) {
+    async endRound(guessed: boolean, messageContext?: MessageContext) {
```
Comment (Brainicism) [inline src/models/game_session.ts:101]: Document newly added parameter

File: src/models/game_session.ts [hunk 2]
```diff
@@ -351,37 +356,36 @@ export default class GameSession {
...
-            await this.endRound(false);
-            this.startRound(guildPreference, message);
+            await this.endRound(false, messageContext);
+            this.startRound(guildPreference, messageContext);
         });
 
         // admin manually 'disconnected' bot from voice channel or misc error
         this.connection.once("error", async (err) => {
             if (!this.connection.channelID) {
                 logger.info(`gid: ${this.textChannel.guild.id} | Bot was kicked from voice channel`);
```
Comment (Brainicism) [inline src/models/game_session.ts:378]: Change to `this.guildID`

File: src/models/game_session.ts [hunk 3]
```diff
@@ -240,7 +246,7 @@ export default class GameSession {
 
         // manage recently played song queue
         if (guildPreference.getShuffleType() === ShuffleType.UNIQUE && Math.min(guildPreference.getLimit(), totalSongs) === this.lastPlayedSongsQueue.length) {
-            logger.info(`${getDebugContext(message)} | Resetting lastPlayedSongsQueue (all ${guildPreference.getLimit()} unique songs played)`);
+            logger.info(`gid: ${this.guildID} | Resetting lastPlayedSongsQueue (all ${guildPreference.getLimit()} unique songs played)`);
```
Comment (Brainicism) [inline src/models/game_session.ts:249]: You can probably overload `getDebugContext` to take in a `MessageContext` and print out this

**Review by Brainicism — CHANGES_REQUESTED** [review]

File: src/models/game_session.ts (hunk 3 above)
Comment (taahamahdi) [reply src/models/game_session.ts:249]: I don't think I can export the same function name twice. An alternative is to make `getDebugContext`'s parameter of type `any`, and then use `instanceof` to decide what to return.

Comment (Brainicism) [reply src/models/game_session.ts:249]: Probably union type

**Review by Brainicism — APPROVED** [review]

**Pull request merged**